hobbit-gui/gui-serverbackend/src/main/java/de/usu/research/hobbit/gui/rest/ExperimentsResources.java

```diff
@@ -65,28 +67,81 @@
         if (Application.isUsingDevDb()) {
             return getDevDb().queryExperiments(ids, challengeTaskId);
         } else {
+            // get user
+            UserInfoBean userInfo = InternalResources.getUserInfoBean(sc);
+            List<SystemBean> userSystems = PlatformControllerClientSingleton.getInstance().requestSystemsOfUser(userInfo.getPreferredUsername());
```

Comment:
I would move this line into the cases in which it is needed, because it represents a communication with the platform controller which (compared to other steps) might take some time.